Fix Lodash isArray deprecation

damassi opened the pull request:
When installing the latest danger, I noticed this in the console output: 

![screen shot 2017-03-29 at 5 15 32 pm](https://cloud.githubusercontent.com/assets/236943/24482317/22adb636-14a5-11e7-84e4-c93020ed2df8.png)

Replaces `lodash.isarray` with native implementation.

DangerCI commented:
<!--
  0 failure: 
  2 warning:  Please add a chan..., Changes were made...
  
  
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please add a changelog entry for your changes.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>Changes were made to package.json, but not to yarn.lock.<br/><i>Perhaps you need to run `yarn install`?</i></td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

macklinu commented on source/platforms/github/GitHubGit.ts:
```diff
@@ -114,7 +113,7 @@ export default async function gitDSLForGitHub(api: GitHubAPI): Promise<GitDSL> {
       // changes to JSON DSLs
 
-      if (isarray(diff.after) && isarray(diff.before)) {
+      if (Array.isArray(diff.after) && Array.isArray(diff.before)) {
```
I think we should be safe using the native `Array.isArray()` instead of that deprecated lodash package. 👍

macklinu commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "danger",
-  "version": "0.14.1",
+  "version": "0.14.2",
```
Could you revert this change please? Unless we are good to ship this in a patch release - cc @orta

macklinu commented on changelog.md:
```diff
@@ -4,10 +4,11 @@
 
 ### 0.14.1
 
+* Moved `@types/chalk` from dependencies to devDependencies - orta
```
Good call, I think this was shipped in 0.14.1.

macklinu commented on changelog.md:
```diff
@@ -4,10 +4,11 @@
 ### 0.14.2
 
-* Moved `@types/chalk` from dependencies to devDependencies - orta
+* Replaced deprecated `lodash.isarray` package with `Array.isArray` - damassi
```
Looking for confirmation from @orta on shipping 0.14.2 or holding off on it for anything else to keep this changelog message here. Otherwise we usually add changelog messages right below the `//  Add your own contribution below` line and then add the version number we are shipping to the changelog as we package up a release.

damassi replied:
Ah, I see -- `danger` said to update the changelog and so I followed the format as it was there. It might be good to provide a little more direction in terms of version numbers, changelog, etc.

macklinu commented:
Forgot to add, Danger commented on the PR requesting that `yarn install` is run (for dependency additions/removals to keep `yarn.lock` in sync with `package.json`). Could you please run that locally and push up a commit with the changes to `yarn.lock`? 🙂

damassi commented:
Updated and rebased. It's odd -- I ran a `yarn install` first thing after pulling down the package, but I had to `rm` it and reinstall packages before it would update.

orta commented:
Ah yeah, the version/changelog thing - again sorry about that, :D

I got stuck on a particularly long PR, which I thought I could get finished by the weekend ( https://github.com/danger/danger-js/pull/194 ) but inevitably I couldn't.

IMO it makes sense to be more verbose in the changelog about saying above the latest released version. 👍

orta commented:
Shipping `0.14.2` sounds 👍

orta commented:
@damassi with respect to the lockfile,  you would have needed to run `yarn remove lodash-isarray` this would have removed the module then only re-created the dependency locks for anything related to it.

In deleting and re-creating it from scratch you've updated hundreds of dependencies in the stack. 

I'm ok with this, after eyeing it over, but ideally next time we should be a bit more strict about smaller lock file changes.

orta commented:
Alright, I've shipped that - and all other PRs as 0.14.2 

I've also updated the notice on the CHANGELOG to be more specific in https://github.com/danger/danger-js/commit/fde4c54d4e01d4f34921aec43624780b2410458e

damassi commented:
Ah sorry about that! I wasn't aware of `yarn remove <package>`. Noted 👍
